### PR TITLE
Upgrade to Font Awesome 6

### LINKED ISSUE #400: Upgrade FontAwesome 4 to FontAwesome 6
In order to use the new optimized assets and have a wider variety of icons, we should upgrade to the latest version of FontAwesome.

**Areas:**
* Views
* Ruby (Helpers)

### PR DESCRIPTION
**Pull request by pixeldesu**
This superseedes #778

### CODE REVIEW

**Review by raccube — APPROVED** [review]

File: app/views/actions/_question.html.haml [hunk 1]
```diff
@@ -13,10 +13,10 @@
       = t("voc.block")
     - if current_user.mod?
       = button_to anonymous_block_path, method: :post, params: { question: question.id, global: true }, class: "dropdown-item" do
-        %i.fa.fa-fw.fa-volume-off
+        %i.fas.fa-fw.fa-user-slash
         = t("voc.block_site_wide")
       %a.dropdown-item{ href: moderation_questions_path(author_identifier: question.author_identifier) }
-        %i.fa.fa-fw.fa-user-secret
+        %i.fas.fa-fw.fa-mask
```
Comment (nilsding) [inline app/views/actions/_question.html.haml]:
```suggestion
        %i.fas.fa-fw.fa-user-secret
```

`fa-mask` looks more like a snout to me

File: app/views/answerbox/_header.html.haml [hunk 2]
```diff
@@ -12,7 +12,7 @@
             = render "actions/question", question: a.question
       %h6.text-muted.media-heading.answerbox__question-user
         - if a.question.author_is_anonymous
-          %i.fa.fa-user-secret{ title: t(".anon_hint") }
+          %i.fas.fa-mask{ title: t(".anon_hint") }
```
Comment (nilsding) [inline app/views/answerbox/_header.html.haml]:
```suggestion
          %i.fas.fa-user-secret{ title: t(".anon_hint") }
```

File: app/views/inbox/_entry.html.haml [hunk 3]
```diff
@@ -7,7 +7,7 @@
       .media-body
         %h6.text-muted.media-heading.answerbox__question-user
           - if i.question.author_is_anonymous
-            %i.fa.fa-user-secret{ title: t('.anon_hint') }
+            %i.fas.fa-mask{ title: t('.anon_hint') }
```
Comment (nilsding) [inline app/views/inbox/_entry.html.haml]:
```suggestion
            %i.fas.fa-user-secret{ title: t('.anon_hint') }
```

**Review by nilsding — APPROVED** [review]

**Review by nilsding — CHANGES_REQUESTED** [review]
fa-mask -> fa-user-secret

**Review by nilsding — APPROVED** [review]

**Pull request merged**